### DISCUSSION TITLE
How to use it, no instructions for beginners.

### DISCUSSION THREAD

**Discussion by bd250124** [pit-ray/win-vind#231] [General]
Can someone tell me how to use this. I saw on the demo that you can access Windows Explore elements with keyboard buttons. I have downloaded portable version and run it. Got the insert mode but that after it? Where to click or what to do, can find any tutorial. I'm bit noob for this xD, but would like to learn xD

**Comment by pit-ray** [comment]
Thanks for your interest.
You can see the tutorial and the concepts in the [Usage](https://pit-ray.github.io/win-vind/usage/) page.

**Comment by kbilsted** [comment]
The demo video shows some impressive window-move-and-resize I can't find in the documentation how to do that myself.

**Reply by pit-ray** [reply]
A list of functions related to window movement and resizing can be found [here](https://pit-ray.github.io/win-vind/cheat_sheet/functions/#window).
Documentation is not enough, but can be made more detailed by your pull request.

**Reply by pit-ray** [reply]
You can refer the default mapping in https://pit-ray.github.io/win-vind/cheat_sheet/defaults/small/#window-openclose .

**Reply by kbilsted** [reply]
I gladly help out, but i cannot explain to others what i dont know myself. for example, the readme shows a configuration file you recommend, but you do not explain what each line does, what they unlock. 

The manual talks about different modes without explaining all the stuff each mode does. you have to read a lot in the guide and the readme just to understand the readme.  For now I've tried using it and god mouse  movement and cursor movement working but i dont understand the windows part. I also have problems with vim and my browser - in vim i prefer vim, and in my browser i have a vim-plugin. 

ultimately, it scares away users. which is a shame because you are making a cool product!

**Reply by pit-ray** [reply]
Thanks for the suggestions for improvement.
The configuration file in the readme is literally an example, not a recommended configuration.

What part of the window operation is confusing to you? Could you be more specific about the details of this?

Open source projects are developed by users as well as maintainers. We would be very grateful if you could give us a pull request to improve the README.

Thanks.

**Comment by kbilsted** [comment]
Hi @pit-ray  

Here are some improvement ideas to the documentation

1. you describe everything as though the reader is a vim expert. i think many users want to get started using vim by learning the motions and using their normal editor, email etc.. this tool is fantastic for learning motions and modes. So to help out new users perhaps explain things a bit more. the video is great in that it gives a teaser of stuff we want. another example is the window resize documentation that refereces winresizer.vim - i am new to vim so i dont know that package. also looking at that package it tells me to do ctrl-e and that doesnt work.


2. for each line in the example config, it should tell what it does. 


3. we are missing step by step guides on how to activate / demo each of the functionalities. for example you give an overview of how to enter certain modes once on the site and thats it. it would be great with for each feature to have a "how to xx" that repeats everything again. The same for the concept of "easyclick".. e.g. 

4. some of the refence documentation is so short that if you haven't programmed it, it is not comprehensible.
 


For the windows resizing features here is an attempt at a howto section. feel free to put the below into the documentation. if you put it in a branch where you fill in the gaps, i can try it out and adjust the documentation further if you want. 



# Howto: Window movements and resizing

Windows support moving windows using windows+cursor keys. but we can do much better! To see this in action ensure that you have started the application in normal mode. Do this by typing `xxx` or look in the config folder at `xxx` and look for the line `...=normal`

then you press `esc+left` to enter 'gui normal mode' 
to split your screen into three windows the first taking up 50% of the screen and the two other windows each taking up 25% of the screen type the following..

* open 3 windows/applications 
* press `xx` to move the active window to the left side of the screen      
* now alt-tab to the next window and press `yy` to move it to the upper right corner 
* finally, alt-tab to the last window and press `zz` to move it to the lower right corner

There are many more functions you can bind to shortcuts, for example tofine-tune the resizing using the commands found in https://pit-ray.github.io/win-vind/cheat_sheet/functions/#window_resizer  

to use such a command 
* open the configuration at `c:\users\...`
* insert the line   `xxxx` 

you can use any hotkey, but i recommend you prefix all your hotkeys with  ctrl-w as this is similar to how many vim users resize their windows.

**Reply by pit-ray** [reply]
Thanks for the great opinions and examples of documentation.

Everything you are saying is correct. In particular, I too have long thought that the quality of documentation and the quantity of documentation should be increased.

However, it is very difficult for win-vind to be a tool for learning motions and modes because even Vim, which has a long history, extensive documentation, rich explanatory articles, and many contributors, takes time for a beginner to understand.

I cannot spend all of my time in life working on this repository, because making this tool does not lead to making any money at all. Also, in registering with well-known package managers such as Chocolatey, winget, scoop, I give the highest priority to fixing bugs for existing users, since it is not recommended by their communities to leave bugs unattended. In other words, I don't have enough resources to write documentation. Thus, it is inevitable that it will take a long time to write the documentation.

If you could help, it would be a tremendous contribution. All you have to do is fork this repository and send a pull request directly to the master branch. You can debug the documentation locally by following [CONTRIBUTING.md](https://github.com/pit-ray/win-vind/blob/master/CONTRIBUTING.md).

Thanks again.

**Reply by kbilsted** [reply]
I have just installed the tool so without some examples to work from i will have a hard time contributing. The last text has gaps can you fill out some of them ?
